### DISCUSSION TITLE
Environment values of spawned process

### DISCUSSION THREAD

**Discussion by pjljvandelaar** [alire-project/alire#1197] [Q&A]
When an alire project spawns a process, does that process get the same environment values as the original process / alire project?

I am asking since `Non_Blocking_Spawn` of GNAT.EXPECT (g-expect.adb) called by gnatcoll-projects.adb crashes on windows on the command `gnatls.exe -v`
while
```
> alr printenv --powershell | Invoke-Expression
> gnatls.exe -v
```
succeeds.

**Comment by mosteo** [comment]
Alire sets the environment that you can see with `alr printenv` for its spawned build processes, and for `alr exec`. We are using the functions in `GNAT.OS_Lib` though.

For other processes unrelated to Ada like `git`, they will see the same environment as `alr` itself.

**Reply by pjljvandelaar** [reply]
So the process spawning `gnatls` from inside my program (started with `alr run`) will have the environment as shown with `alr printenv`?

**Reply by mosteo** [reply]
We have a little inconsistency here now that I've inspected the code but yes, as long as you don't use `alr run --skip-build`, the environment will have been set during the build for any subsequent spawn.
